### PR TITLE
Move common helper functions to pkg test

### PR DESCRIPTION
**Pull request by Fredy-Z**
These two functions `MakeK8sNamePrefix` and `GetBaseFuncName` can be used for both `Serving` and `Eventing` E2E test, so move them here for sharing and simplicity.

### CODE REVIEW

File: test/helpers/data.go [hunk 1]
```diff
@@ -47,3 +48,30 @@ func AppendRandomString(prefix string) string {
...
+func MakeK8sNamePrefix(s string) string {
+	var sb strings.Builder
+	newToken := false
+	for _, c := range s {
+		if !(unicode.IsLetter(c) || unicode.IsNumber(c)) {
+			newToken = true
+			continue
+		}
+		if sb.Len() > 0 && (newToken || unicode.IsUpper(c)) {
+			sb.WriteRune('-')
```
Comment (srinivashegde86) [inline test/helpers/data.go]: use the const

File: test/helpers/data.go [hunk 2]
```diff
@@ -47,3 +48,30 @@ func AppendRandomString(prefix string) string {
...
+		}
+		sb.WriteRune(unicode.ToLower(c))
+		newToken = false
+	}
+	return sb.String()
+}
+
+// GetBaseFuncName returns the baseFuncName parsed from the fullFuncName.
+// eg. test/e2e.TestMain will return TestMain.
+func GetBaseFuncName(fullFuncName string) string {
```
Comment (srinivashegde86) [inline test/helpers/data.go]: can we add ut for these?

Comment (Fredy-Z) [reply test/helpers/data.go]: Added.

File: test/helpers/data.go (hunk 1 above)
Comment (Fredy-Z) [reply test/helpers/data.go]: Used const `dash`.

File: test/helpers/data.go [hunk 3]
```diff
@@ -52,6 +52,7 @@ func AppendRandomString(prefix string) string {
 // MakeK8sNamePrefix converts each chunk of non-alphanumeric character into a single dash
 // and also convert camelcase tokens into dash-delimited lowercase tokens.
 func MakeK8sNamePrefix(s string) string {
+	const dash = '-'
```
Comment (srinivashegde86) [inline test/helpers/data.go]: there is already a const in the file called sep - https://github.com/knative/pkg/blob/master/test/helpers/data.go#L28

File: test/helpers/data_test.go [hunk 4]
```diff
@@ -17,16 +17,49 @@ limitations under the License.
...
 import (
-	"fmt"
 	"regexp"
+	"testing"
 )
 
 var matcher = regexp.MustCompile("abcd-[a-z]{8}")
 
-func ExampleAppendRandomString() {
+func TestAppendRandomString(tst *testing.T) {
```
Comment (srinivashegde86) [inline test/helpers/data_test.go]: use t instead of tst. for the tests

File: test/helpers/data_test.go [hunk 5]
```diff
@@ -17,16 +17,49 @@ limitations under the License.
...
 	const s = "abcd"
 	t := AppendRandomString(s)
 	o := AppendRandomString(s)
-	fmt.Println(matcher.MatchString(t), matcher.MatchString(o), o != t)
-	// Output: true true true
+	if !matcher.MatchString(t) || !matcher.MatchString(o) || o == t {
+		tst.Fatal()
```
Comment (srinivashegde86) [inline test/helpers/data_test.go]: t.Fatalf(<msg>). It should fial with an err message

File: test/helpers/data_test.go [hunk 6]
```diff
@@ -17,16 +17,49 @@ limitations under the License.
...
+}
+
+func TestMakeK8sNamePrefix(tst *testing.T) {
+	testCases := map[string]string{
+		"abcd123":  "abcd123",
+		"AbCdef":   "ab-cdef",
+		"ABCD":     "a-b-c-d",
+		"aBc*ef&d": "a-bc-ef-d",
+	}
+	for k := range testCases {
```
Comment (srinivashegde86) [inline test/helpers/data_test.go]: for _,k := range tc {

File: test/helpers/data_test.go [hunk 7]
```diff
@@ -17,16 +17,49 @@ limitations under the License.
...
 	o := AppendRandomString(s)
-	fmt.Println(matcher.MatchString(t), matcher.MatchString(o), o != t)
-	// Output: true true true
+	if !matcher.MatchString(t) || !matcher.MatchString(o) || o == t {
+		tst.Fatal()
+	}
+}
+
+func TestMakeK8sNamePrefix(tst *testing.T) {
+	testCases := map[string]string{
```
Comment (srinivashegde86) [inline test/helpers/data_test.go]: Use the table driven tests format

tc := struct {
want string
got string 
}{ 
{"abcd123": "abcd123"},
...
}

File: test/helpers/data_test.go [hunk 8]
```diff
@@ -17,16 +17,49 @@ limitations under the License.
...
+		expected := testCases[k]
+		actual := MakeK8sNamePrefix(k)
+		if expected != actual {
+			tst.Fatalf("Expect %q but actual is %q", expected, actual)
+		}
+	}
+}
+
+func TestGetBaseFuncName(tst *testing.T) {
+	testCases := map[string]string{
```
Comment (srinivashegde86) [inline test/helpers/data_test.go]: use table driven tests format

File: test/helpers/data_test.go [hunk 9]
```diff
@@ -17,16 +17,49 @@ limitations under the License.
...
+		"test/e2e.TestMain": "TestMain",
+		"e2e.TestMain":      "TestMain",
+		"test/TestMain":     "TestMain",
+		"TestMain":          "TestMain",
+	}
+	for k := range testCases {
```
Comment (srinivashegde86) [inline test/helpers/data_test.go]: for _,k := range tc {
got := GetBaseFuncName(k.name)
if k.want != got {
t.Fatalf(<msg>)
}
}

**Pull request merged**